### ISSUE TITLE
[1.3.0-RC1] concurrent runs intermittently cause locking failures

### ISSUE THREAD

**Issue by SethTisue** [sbt/sbt#4723]
I was running two instances of the Scala community build at the same time on my laptop, and in both runs, I had several projects fail with this error:

```
[akka] [error] java.io.FileNotFoundException: /Users/tisue/Library/Caches/Coursier/v1/.structure.lock (No such file or directory)
[akka] [error] 	at java.io.FileOutputStream.open0(Native Method)
[akka] [error] 	at java.io.FileOutputStream.open(FileOutputStream.java:270)
[akka] [error] 	at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
[akka] [error] 	at java.io.FileOutputStream.<init>(FileOutputStream.java:162)
[akka] [error] 	at lmcoursier.internal.shaded.coursier.paths.CachePath.withStructureLock(CachePath.java:112)
[akka] [error] 	at lmcoursier.internal.shaded.coursier.cache.CacheLocks$.withStructureLock(CacheLocks.scala:20)
[akka] [error] 	at lmcoursier.internal.shaded.coursier.cache.CacheLocks$.withLockOr(CacheLocks.scala:34)
[akka] [error] 	at lmcoursier.internal.shaded.coursier.cache.FileCache.$anonfun$download$25(FileCache.scala:429)
[akka] [error] 	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:658)
[akka] [error] 	at scala.util.Success.$anonfun$map$1(Try.scala:255)
[akka] [error] 	at scala.util.Success.map(Try.scala:213)
[akka] [error] 	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[akka] [error] 	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[akka] [error] 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[akka] [error] 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[akka] [error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[akka] [error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[akka] [error] 	at java.lang.Thread.run(Thread.java:748)
[akka] [error] (update) java.io.FileNotFoundException: /Users/tisue/Library/Caches/Coursier/v1/.structure.lock (No such file or directory)
[akka] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
```

the affected projects were airframe, akka, coursier, github4s

**Comment by SethTisue** [comment]
log of the entire run: https://gist.github.com/SethTisue/af4544a5aa6eb263eef5a954a7777466

**Comment by eed3si9n** [comment]
@SethTisue Thanks for the report!
/cc @alexarchambault

**Comment by eed3si9n** [comment]
Likely fixed by #4845
